pkgs/by-name/fy/fyne/package.nix

```
@@ -0,0 +1,72 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+
+  dbus,
+  libGL,
+  libX11,
+  libXcursor,
+  libXinerama,
+  libXi,
+  libXrandr,
+  libXxf86vm,
+  pkg-config,
+}:
+
+let
+  version = "2.4.4";
+in buildGoModule {
+  pname = "fyne";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "fyne-io";
+    repo = "fyne";
+    rev = "v${version}";
+    hash = "sha256-6gDHURcf4dVmaiXleeRKogAtCAgcHU437TjEHhfGdj8=";
+  };
+
+  vendorHash = "sha256-0P2YqXq+c20F2hVkRcvwV5PpqUoZ3PnRIvTHt/qAMPA=";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    libGL
+    libX11
+    libXcursor
+    libXinerama
+    libXi
+    libXrandr
+    libXxf86vm
+  ];
+
+  checkInputs = [
+    dbus
+  ];
+
+  doCheck = true;
+
+  preCheck = ''
+    OLD_DIR=$PWD
+    OLD_GOPATH=$GOPATH
+    export HOME=$(mktemp -d)
+    export GOPATH=$(mktemp -d)
+    cd $HOME
+  '';
+
+  preInstall = ''
+    cd $OLD_DIR
+    export GOPATH=$OLD_GOPATH
```

```suggestion
    [[ -n $OLD_GOPATH ]] && export GOPATH=$OLD_GOPATH
```

Or someone will have problem disabling `doCheck`.

---

Good catch. I've removed the redefining of GOPATH, but I've used a similar paradigm for the `cd` command above it for the same reason. In my local tests disabling checks caused an error like you said.